pydantic/typing.py

```diff
@@ -76,9 +77,14 @@ def get_all_type_hints(obj: Any, globalns: Any = None, localns: Any = None) -> A
         return get_type_hints(obj, globalns, localns, include_extras=True)
 
 
+_T = TypeVar('_T')
+
 AnyCallable = TypingCallable[..., Any]
 NoArgAnyCallable = TypingCallable[[], Any]
 
+# workaround for https://github.com/python/mypy/issues/9496
+AnyArgAnyCallable = TypingCallable[..., _T]
```

Comment:
```suggestion
AnyArgTCallable = TypingCallable[..., _T]
```
Or something, since its return type is paramaterised.